src/types.js

```diff
@@ -394,13 +549,18 @@ encode.MACSTRING = function(str, encoding) {
                 return undefined;
             }
         }
-
-        result.push(c);
+        result[i] = c;
```

Comment:
@Harbs After searching for the best one in term of performance it looks like it's more complicated than that…

> Usually there is no reason not to use push - the method is there for exactly that operation and makes some code easier to read. While a few people think one version is faster than the other, both are equally optimized in browsers. See Why is array.push sometimes faster than array[n] = value? and Using the push method or .length when adding to array? - results vary so wide that it's actually irrelevant. Use what is better to understand.

from http://stackoverflow.com/a/21034712/1345244

> Actually push is just as fast as property assignment. Good measurements are rare, but those that are done properly show varying results across different browser engine versions - changing rapidly and unexpected. See Appending to array, Why is array.push sometimes faster than array[n] = value? and Is there a reason JavaScript developers don't use Array.push()? - the conclusion is that you should use what is most readable / appropriate for your use case, not what you think could be faster.
from http://stackoverflow.com/a/18477047/1345244

Also, I know that AirBnB did a good job with this public style guide https://github.com/airbnb/javascript/blob/master/es5/README.md#arrays and use `.push()`.
In the end maybe the performance is not really easy to compare here and readability is more important?